src/ClockPanel.tsx

```diff
@@ -19,18 +20,57 @@ export function getTimeZoneNames(): string[] {
 }
 
 class UnthemedClockPanel extends PureComponent<Props, State> {
-  timerID?: any;
+  timerID?: any = 0;
   state = { now: this.getTZ(), timezone: '' };
+  subscription?: Unsubscribable;
 
   componentDidMount() {
+    this.subscription = this.props.eventBus.subscribe({ type: 'refresh' } as any, this.onPanelRefresh);
```

Comment:
this should not really be needed, if the props.data has changed there has been a refresh

Reply:
There is no query, so the update never happens

Reply:
Took @torkelo suggestion for a spin and it seems to work like a charm. So I have changed the code accordingly.